pkg/templates/manifest.go

```diff
@@ -97,8 +120,19 @@ data:
       - mountPath: /tmp/results
         name: results
         readOnly: false
+{{- if .SSHKey }}
+      - mountPath: /root/.ssh
+        name: sshkey-vol
+{{- end}}
       tolerations:
         - operator: "Exists"
+{{- if .SSHKey }}
+    extra-volumes:
+    - name: sshkey-vol
+      secret:
+        secretName: ssh-key
+        defaultMode: 256
```

Comment:
this is a weird mode. `--w-r-xrw-`. I don't think this'll work, usually `600` is the right mode for SSH keys.

Reply:
Its a little wonky in the documentation because they go back and forth but it seems to require decimal notation:
> Note that the JSON spec doesn’t support octal notation, so use the value 256 for 0400 permissions. If you use yaml instead of json for the pod, you can use octal notation to specify permissions in a more natural way.

If you look at their examples though, even when using yaml they use the decimal notation (so 256->0400 which is sufficient for this purpose)

https://kubernetes.io/docs/concepts/configuration/secret/

Reply:
huh, TIL